### PR TITLE
Spawning more than 3 traitors, or more than 2 vampires, cost a bit more threat.

### PR DESCRIPTION
**Pull request by ShiftyRail**
Right now, it only ever costs 10 threat to spawn traitors, but it scales with population and threat.
This means you can have as much as 7 (seven) traitors for just 10 threat. 7 traitors is a very destructive and should cost the threat required.

Moreover, given that traitors refund threat when gibbed, they can refund threat for **other things** than themselves.

Let's assume we roll 50 threat. We roll traitor (10 threat) and vampire (25 threat).
We spawn 5 traitors and 3 vampires. 2 vampires get ashed in the chapel and one traitor e-implants himself. That's 30 threat refunded.

We are back 50 threat - enough for midround traitors, ninja, and so on - but we still have 4 traitors on the run and one vampire.

Alternative PR to #22729.

The motivation is the same : too many gunked rounds, too much traitors at any given time.

:cl:
- tweak: It will now cost more threat to spawn more than 3 traitors and more than 2 vampires.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
That's better
Only problem is that vampire costs will rack up rapidly and a fourth vampire means 30 threat gone

File: code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm [hunk 1]
```diff
@@ -28,6 +30,11 @@
 		var/datum/role/traitor/newTraitor = new
 		newTraitor.AssignToRole(M.mind,1)
 		newTraitor.Greet(GREET_ROUNDSTART)
+		// Above 3 traitors, we start to cost a bit more.
+		if (i > max_traitors && (mode.threat_level > additional_cost))
```
Comment (boy2mantwicethefam) [inline code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm]: Threat level 5 (out of 100) is basically extended already, I don't know the purpose of having this check here, ditto for vampires

File: code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm [hunk 2]
```diff
@@ -14,6 +14,8 @@
 	required_candidates = 1
 	weight = 5
 	cost = 10
+	var/max_traitors = 3
```
Comment (boy2mantwicethefam) [inline code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm]: A non-concern but max_traitors is non-indicative of what this does, maybe change it to threshold_traitors? Ditto for vampires

File: code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm]: I was certain `threat_level`  was the remaining threat. It's actually `threat`, my mistake.

**Comment by jknpj** [comment]
remove two starting vampires when? knowing FOR SURE that there's another vampire after you find the first one is METATERRIBLE

**Comment by ShiftyRail** [comment]
Not that big of a problem for me, people will always behave like there is more vampires anyways.

**Comment by Kurfursten** [comment]
It should probably have a reduced base cost and set it to only require 1 candidate though, that way it *could* fire with only 1 vamp.

Needs CL

**Comment by Caboozles** [comment]
If by reduced cost you mean that if only one fires it’s reduced, then sure. But I am very ready for less than Unlimited Vampire Works.

**Comment by Kurfursten** [comment]
The way vampire currently works is that it costs a flat 25 no matter how many are made
Shifty is changing it so that it costs 25 base + 15 for each additional vampire past 2
Instead of that, I suggest it cost 15 base (for one vampire) spend 15 additional for each additional vampire.

**Comment by DamianX** [comment]
Kurfursten's suggestion makes much more sense, please do that instead

**Comment by ShiftyRail** [comment]
I'll do that.

**Pull request merged**